include/SDL3/SDL_intrin.h

```diff
@@ -108,27 +113,39 @@ _m_prefetch(void *__P)
 #endif
 
 #if defined(__loongarch_sx) && !defined(SDL_DISABLE_LSX)
-#include <lsxintrin.h>
+# define SDL_LSX_INTRINSICS 1
+# include <lsxintrin.h>
 #endif
+
 #if defined(__loongarch_asx) && !defined(SDL_DISABLE_LASX)
-#include <lasxintrin.h>
+# define SDL_LASX_INTRINSICS 1
+# include <lasxintrin.h>
 #endif
+
 #if defined(__x86_64__) || defined(_M_X64) || defined(__i386__) || defined(_M_IX86)
-#if (defined(__AVX__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_AVX)
-#include <immintrin.h>
-#endif
-#if (defined(__MMX__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_MMX)
-#include <mmintrin.h>
-#endif
-#if (defined(__SSE__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_SSE)
-#include <xmmintrin.h>
-#endif
-#if (defined(__SSE2__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_SSE2)
-#include <emmintrin.h>
-#endif
-#if (defined(__SSE3__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_SSE3)
-#include <pmmintrin.h>
-#endif
-#endif /**/
+# if (defined(__AVX__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_AVX)
+#  define SDL_AVX_INTRINSICS
+#  include <immintrin.h>
+# endif
+# if (defined(__MMX__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_MMX)
+#  define SDL_MMX_INTRINSICS 1
+#  include <mmintrin.h>
+# endif
+# if ((defined(_MSC_VER) && defined(_M_IX86) && _M_IX86_FP == 1) || defined(_M_X64) || defined(__SSE__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_SSE)
```

Comment:
If we really need it (see my other comment about it), should we check `_M_IX86_FP` as `== 1` or `>= 1` ?

Reply:
You're right, SSE is enabled when ` (_M_IX86_FP == 1 || _M_IX86_FP == 2 )`.

```suggestion
# if ((defined(_MSC_VER) && defined(_M_IX86) && (_M_IX86_FP == 1 || _M_IX86_FP == 2 )) || defined(_M_X64) || defined(__SSE__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_SSE)
```

Reply:
Looking at VS2022 headers, SSEx and even AVX doesn't seem to need command line switch activation (only MMX doesn't seem to be supported with X64), so I suggest the following: Can you confirm?

```diff
diff --git a/include/SDL3/SDL_intrin.h b/include/SDL3/SDL_intrin.h
index a01dfbe..1583238 100644
--- a/include/SDL3/SDL_intrin.h
+++ b/include/SDL3/SDL_intrin.h
@@ -36,39 +36,24 @@
 #ifdef __clang__
 /* As of Clang 11, '_m_prefetchw' is conflicting with the winnt.h's version,
    so we define the needed '_m_prefetch' here as a pseudo-header, until the issue is fixed. */
-
 #ifndef __PRFCHWINTRIN_H
 #define __PRFCHWINTRIN_H
-
 static __inline__ void __attribute__((__always_inline__, __nodebug__))
 _m_prefetch(void *__P)
 {
   __builtin_prefetch (__P, 0, 3 /* _MM_HINT_T0 */);
 }
-
 #endif /* __PRFCHWINTRIN_H */
 #endif /* __clang__ */
 #include <intrin.h>
-#ifndef _WIN64
-#ifndef __MMX__
-#define __MMX__
-#endif
-#endif
-#ifndef __SSE__
-#define __SSE__
-#endif
-#ifndef __SSE2__
-#define __SSE2__
-#endif
-#ifndef __SSE3__
-#define __SSE3__
-#endif
+
 #elif defined(__MINGW64_VERSION_MAJOR)
 #include <intrin.h>
 #if defined(__ARM_NEON) && !defined(SDL_DISABLE_NEON)
 #  define SDL_NEON_INTRINSICS 1
 #  include <arm_neon.h>
 #endif
+
 #else
 /* altivec.h redefining bool causes a number of problems, see bugs 3993 and 4392, so you need to explicitly define SDL_ENABLE_ALTIVEC to have it included. */
 #if defined(__ALTIVEC__) && defined(SDL_ENABLE_ALTIVEC)
@@ -123,25 +108,25 @@ _m_prefetch(void *__P)
 #endif
 
 #if defined(__x86_64__) || defined(_M_X64) || defined(__i386__) || defined(_M_IX86)
-# if (defined(__AVX__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_AVX)
+# if (defined(_MSC_VER) || defined(__AVX__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_AVX)
 #  define SDL_AVX_INTRINSICS
 #  include <immintrin.h>
 # endif
-# if (defined(__MMX__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_MMX)
+# if ((defined(_MSC_VER) && !defined(_M_X64)) || defined(__MMX__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_MMX)
 #  define SDL_MMX_INTRINSICS 1
 #  include <mmintrin.h>
 # endif
-# if ((defined(_MSC_VER) && defined(_M_IX86) && _M_IX86_FP == 1) || defined(_M_X64) || defined(__SSE__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_SSE)
+# if (defined(_MSC_VER) || defined(__SSE__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_SSE)
    /* x86 MSVC defines _M_IX86_FP == 1 when compiled with /arch:SSE */
 #  define SDL_SSE_INTRINSICS 1
 #  include <xmmintrin.h>
 # endif
-# if ((defined(_MSC_VER) && defined(_M_IX86) && _M_IX86_FP >= 1) || defined(_M_X64) || defined(__SSE2__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_SSE2)
+# if (defined(_MSC_VER) || defined(__SSE2__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_SSE2)
    /* x86 MSVC defines _M_IX86_FP == 2 when compiled with /arch:SSE2 */
 #  define SDL_SSE2_INTRINSICS 1
 #  include <emmintrin.h>
 # endif
-# if (defined(_M_X64) || defined(__SSE3__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_SSE3)
+# if (defined(_MSC_VER) || defined(__SSE3__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_SSE3)
    /* x86 MSVC does not provide macro's to detect SSE3/SSE4 */
 #  define SDL_SSE3_INTRINSICS 1
 #  include <pmmintrin.h>
```

Reply:
Yes, I am seeing this too on vs2019.
I'm creating a small (automation) test atm.
I also think these conditions can be simplified even further. But I let ci decide on that :].

Because vs2010 is the oldest supported compiler, I don't think we need to be extra careful for avoiding the avx immintrin.h header.

We probably need extra checks when we introduce `SDL_AVX512F_INTRINSICS`.

Reply:
OK, applied my patch. Simplify further if you can.

Waiting for your automation test before merging.